src/primitive_crypto.cc

```diff
@@ -121,31 +159,52 @@ PRIMITIVE(siphash_get) {
   return result;
 }
 
-AesCbcContext::AesCbcContext(
+AesContext::AesContext(
     SimpleResourceGroup* group,
-    const uint8* key,
-    const uint8* iv,
-    bool encrypt)
-  : SimpleResource(group) {
+    const Blob* key,
+    bool encrypt) : SimpleResource(group) {
   mbedtls_aes_init(&context_);
   if (encrypt) {
-    mbedtls_aes_setkey_enc(&context_, key, 256);  // 32 byte key.
+    mbedtls_aes_setkey_enc(&context_, key->address(), key->length() * 8); 
   } else {
-    mbedtls_aes_setkey_dec(&context_, key, 256);  // 32 byte key.
+    mbedtls_aes_setkey_dec(&context_, key->address(), key->length() * 8);
```

Comment:
```suggestion
    mbedtls_aes_setkey_dec(&context_, key->address(), key->length() * BYTE_BIT_SIZE);
```

Reply:
Changed from magic number to named definition.